src/lib/components/Games/HolidayGame.svelte

```diff
@@ -0,0 +1,131 @@
+<script lang="ts">
+	import { browser } from '$app/environment';
+	import { holidayLists } from '$lib/assets/holidayLists';
+	import Board from '$lib/components/Board.svelte';
+	import GameSummary from '$lib/components/GameSummary.svelte';
+	import Timer from '$lib/components/Timer.svelte';
+	import { gameProgress } from '$lib/store';
+	import { formWord, makeWordUsed, type LetterData } from '$lib/utils/letterDataUtils';
+	import { createLetterData } from '$lib/utils/letterDataUtils';
+	import { checkIfValidWord } from '$lib/utils/wordUtils';
+
+	let now = new Date();
+	const wordList = holidayLists.find((list) => {
+		return now.getMonth() == list.date.getMonth() && now.getDate() == list.date.getDate();
+	})?.words;
+
+	if (wordList == undefined) {
+		throw new Error("No holiday found for today's date");
+	}
+
+	wordList.push(''); // finite list will break the reactive statement for unplayedLetters due to the array going out of bounds
+
+	gameProgress.set(0);
+
+	const wordsToSolve = wordList.length - 1;
+	let solvedWords: string[] = [];
+
+	let wordToGuess = wordList[0];
+	let playedLetters: LetterData[] = createLetterData([wordToGuess[0]], true);
+	let unplayedLetters: LetterData[];
+
+	let gameComplete = false;
+	let wasGameAlreadyPlayed = false;
+
+	let elapsedTime = 0;
+
+	let wrongGuess = false;
+
+	// update the board when we get a new word to guess
+	$: unplayedLetters = createLetterData([...wordToGuess.substring(1)], false, true);
+
+	if (browser && window.localStorage.getItem('lastPlayed') == now.toDateString()) {
+		gameComplete = true;
+		wasGameAlreadyPlayed = true;
+	}
+
+	const moveLetter = (toPlayed: boolean, index: number) => {
+		let letter: LetterData;
+		if (index < 0) {
+			return;
+		}
```

Comment:
Style nitpick, early returns should be one line
```suggestion
		if (index < 0) return;
```